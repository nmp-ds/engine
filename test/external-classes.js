import { setup } from './_helpers.js';
import { describe, expect, test } from 'vitest';

setup({ externalizeClasses: true, externalClasses: ["mt-32", "mb-32"] });

describe('external classes', () => {
  test('externalized classes of mt-32 and mb-32 should be excluded from the css', async ({ uno }) => {
    const classes = ['mt-16', 'mt-32', 'mb-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('externalized class of mt-32 should be excluded from the css', async ({ uno }) => {
    const classes = ['mt-16', 'mt-32', 'mb-24'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('externalized class of mt-32 should be included if used with modifier', async ({ uno }) => {
    const classes = ['sm:mt-32', 'md:mt-32', 'active:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
});
