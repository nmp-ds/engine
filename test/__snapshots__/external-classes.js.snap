// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`external classes > externalized class of mt-32 should be excluded from the css 1`] = `
"/* layer: default */
.mb-24{margin-bottom:2.4rem;}
.mt-16{margin-top:1.6rem;}"
`;

exports[`external classes > externalized class of mt-32 should be included if used with modifier 1`] = `
"/* layer: default */
.active\\\\:mt-32:active{margin-top:3.2rem;}
@media (min-width: 480px){
.sm\\\\:mt-32{margin-top:3.2rem;}
}
@media (min-width: 768px){
.md\\\\:mt-32{margin-top:3.2rem;}
}"
`;

exports[`external classes > externalized classes of mt-32 and mb-32 should be excluded from the css 1`] = `
"/* layer: default */
.mt-16{margin-top:1.6rem;}"
`;
